extensions/desktop-common/desktop-init

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -e
+
+#################
+# Launcher init #
+#################
+
+# On Fedora $SNAP is under /var and there is some magic to map it to /snap.
+# # We need to handle that case and reset $SNAP
+SNAP="${SNAP/\/var\/lib\/snapd/}"
+
+SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+
+# shellcheck source=/dev/null
+. "$SNAP_USER_DATA/.last_revision" 2>/dev/null || true
+if [ "$SNAP_DESKTOP_LAST_REVISION" = "$SNAP_REVISION" ]; then
+  SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="false"
+fi
+
+# Set $REALHOME to the users real home directory
+REALHOME="$(getent passwd $UID | cut -d ':' -f 6)"
+
+# If the user has modified their user-dirs settings, force an update
+if [[ -f "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum" && -f "$XDG_CONFIG_HOME/user-dirs.locale.md5sum" ]]; then
+  if [[ "$(md5sum < "$REALHOME/.config/user-dirs.dirs")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum")" ||
+        "$(md5sum < "$REALHOME/.config/user-dirs.locale")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.locale.md5sum")" ]]; then
+    SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+  fi
+fi
+
+# Create config folder
+mkdir -p "$XDG_CONFIG_HOME" && chmod 700 "$XDG_CONFIG_HOME"
```

Comment:
At this point in the command chain, `XDG_CONFIG_HOME` hasn't been defined yet, and this results in:

    mkdir: cannot create directory '': No such file or directory